### DISCUSSION TITLE
[Actions-Variable] How do I get the &Parent from &Parent?

### DISCUSSION THREAD

**Discussion by merhovon** [codefori/vscode-ibmi#865] [Q&A]
Hello all,

I need the parent folder for a command, how can I determine it?

My current path (IFS):
/home/XXXX/ABC/QCBLPGM/HALLOWELT.CBLLE

The parameter &Parent returns me QCBLPGM but I still need ABC and possibly also the folder above. !!!Only the folder name!!!

**Comment by worksofliam** [comment]
You likely should be using `&HOME` which refers to the current working directory (so perhaps `/home/XXXX/ABC/`)

**Comment by merhovon** [comment]
@worksofliam 
I know the &HOME variable returns the whole path but I need the &Parent variable to return only a part of it.
I want to be able to stagger &Parent so that I can determine 'ABC' or 'XXXX' as a single value.

**Reply by worksofliam** [reply]
@merhovon What do you propose the variable name be for the 'parent of the parent'?

**Reply by merhovon** [reply]
@worksofliam 
Maybe superordinate? Is it possible that this variable can determine values dynamically? The idea is that if I call the new variable with itself, that I get the directory above it.

Here is an example:

My current path (IFS):
/home/XXXX/ABC/QCBLPGM/HALLOWELT.CBLLE

&superordinate -> ABC
&superordinate(&superordinate)  -> XXXX
&superordinate(&superordinate(&superordinate)) -> home

**Reply by merhovon** [reply]
@worksofliam 
Is there any planning yet as to when this will be implemented?

**Reply by worksofliam** [reply]
@merhovon No plans from me at the moment. 

Sorry, I have a lot of stuff on my plate at the moment. When I have more time, perhaps we could add this kind of stuff to Actions.
